my-app/src/app/core/ui/header/header.component.ts

```diff
@@ -1,18 +1,30 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { AuthService } from '../../services/auth.service';
 import { User } from '../../../shared/models/user.model';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'cp-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss']
 })
-export class HeaderComponent implements OnInit {
+export class HeaderComponent implements OnInit, OnDestroy {
   public user!: User;
+  private unsubscribe = new Subject();
 
   constructor(public authService: AuthService) { }
 
   ngOnInit(): void {
-    this.user = this.authService.getUserInfo();
+    this.authService.user$.pipe(takeUntil(this.unsubscribe)).subscribe(user => {
```

Comment:
I would use it with async pipe, it will be easier and cleaner imo:

`this.user$ = this.authService.user$;` , in template `*ngIf="user$ | async as user"`